cli/src/pcluster/validators/database_validators.py

```diff
@@ -0,0 +1,128 @@
+            self._add_failure(
+                f"Invalid URI specified. Please remove any leading / at the beginning of the provided URI ('{uri}')",
+                FailureLevel.ERROR,
+            )
+            return False
+        return True
+
+    def _check_scheme(self, uri_parse: ParseResult) -> bool:
+        try:
+            scheme = uri_parse.scheme
+        except ValueError as e:
+            self._add_failure("Invalid URI specified. " + str(e), FailureLevel.ERROR)
+            return False
+        if scheme:
+            self._add_failure(
+                f"Invalid URI specified. Please do not provide a scheme ('{scheme}://')",
+                FailureLevel.ERROR,
+            )
+            return False
+        return True
+
+    def _check_netloc(self, uri: str, uri_parse: ParseResult) -> bool:
+        try:
+            netloc = uri_parse.netloc
+        except ValueError as e:
+            self._add_failure("Invalid URI specified. " + str(e), FailureLevel.ERROR)
+            return False
+        if not netloc:
+            self._add_failure(
+                f"Invalid URI specified. Please review the provided URI ('{uri}')",
+                FailureLevel.ERROR,
+            )
+            return False
+        return True
+
+    def _check_port(self, uri_parse: ParseResult) -> bool:
+        default_mysql_port = 3306
+        try:
+            port = uri_parse.port
+        except ValueError as e:
+            self._add_failure("Invalid URI specified. " + str(e), FailureLevel.ERROR)
+            return False
+        if not port:
+            self._add_failure(
+                f"No port specified in the URI. Assuming the use of port {default_mysql_port}",
+                FailureLevel.WARNING,
+            )
+        return True
+
+
+class SlurmdbdStorageParametersValidator(Validator):
+    """Slurmdbd StorageParameters validator"""
+
+    def _validate(self, slurmdbd_storage_parameters: dict):
+        """Validate slurmdbd StorageParameters."""
+        if not slurmdbd_storage_parameters:
+            self._add_failure(
+                "SlurmdbdStorageParameters not provided. This may prevent database "
+                "server identity verification from slurmdbd daemon.",
+                FailureLevel.WARNING,
+            )
+            return
+        config_keys = slurmdbd_storage_parameters.keys()
+        if "SSL_CA" not in config_keys:
+            self._add_failure(
+                "'SSL_CA' Slurmdbd StorageParameter not provided. This may prevent database "
+                "server identity verification from slurmdbd daemon.",
+                FailureLevel.WARNING,
+            )
+        for key, value in slurmdbd_storage_parameters.items():
+            if key not in SLURMDBD_STORAGE_PARAMETERS:
+                self._add_failure(
+                    f"'{key}' not available as Slurmdbd StorageParameter. The provided parameter will be ignored.",
+                    FailureLevel.INFO,
+                )
```

Comment:
why not failing? a typo might go unnoticed otherwise

Reply:
We need to discuss about this implementation. I did not create a schema for the `StorageParameter` parameter because I think we should not take ownership of that parameter (and in the longer run we should have a more generalized solution to allow users to customize the Slurm configuration).

Regarding the FailureLevel, I guess I can increase it to ERROR.

Reply:
Done.